vscode/src/client/analyzerClient.ts

```diff
@@ -83,19 +83,25 @@ export class AnalyzerClient {
     this.outputChannel.appendLine(`server args:`);
     this.getKaiRpcServerArgs().forEach((arg) => this.outputChannel.appendLine(`   ${arg}`));
 
-    this.kaiRpcServer = spawn(this.getKaiRpcServerPath(), this.getKaiRpcServerArgs(), {
+    const kaiRpcServer = spawn(this.getKaiRpcServerPath(), this.getKaiRpcServerArgs(), {
       cwd: serverCwd,
       env: this.getKaiRpcServerEnv(),
     });
+    this.kaiRpcServer = kaiRpcServer;
+
+    const pid = await new Promise<number | undefined>((resolve, reject) => {
+      kaiRpcServer.on("spawn", () => {
+        this.outputChannel.appendLine(
+          `kai rpc server has been spawned! [${this.kaiRpcServer?.pid}]`,
+        );
+        resolve(this.kaiRpcServer?.pid);
+      });
 
-    this.kaiRpcServer.on("spawn", () => {
-      this.outputChannel.appendLine(`kai rpc server has been spawned! [${this.kaiRpcServer?.pid}]`);
-    });
-
-    this.kaiRpcServer.on("error", (err) => {
-      this.outputChannel.appendLine(
-        `[error] - error in process[${this.kaiRpcServer?.spawnfile}]: ${err}`,
-      );
+      kaiRpcServer.on("error", (err) => {
+        const message = `error in process[${this.kaiRpcServer?.spawnfile}]: ${err}`;
+        this.outputChannel.appendLine(`[error] - ${message}}`);
+        reject();
```

Comment:
`reject`  will break the flow - we should probably add `catch` block that would i.e. set the server status (I believe this kind of problems motivated https://github.com/konveyor/editor-extensions/issues/135)

Reply:
That is part of it for sure.  But really the server's `ServerState` should be tracked much more explicitly than it is now.  There are some gaps in potential.  I can address those in another followup PR.